src/scripts/main.js

```diff
@@ -1 +1,90 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = () => {
+    clearTimeout(timer);
+    document.removeEventListener('click', clickHandler);
+
+    resolve('First promise was resolved');
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  const timer = setTimeout(() => {
+    document.removeEventListener('click', clickHandler);
```

Comment:
The 'clickHandler' function is removed from the 'click' event inside the setTimeout callback, but it should also be removed from the 'contextmenu' event to prevent potential memory leaks or unexpected behavior.